enterprise_access/apps/bffs/handlers.py

```diff
@@ -69,29 +73,97 @@ def load_and_process(self):
         The method in this class simply calls common learner logic to ensure the context is set up.
         """
         try:
+            # Transform enterprise customer data
+            self.transform_enterprise_customers()
+
             # Retrieve and process subscription licenses. Handles activation and auto-apply logic.
-            # TODO: retrieve enterprise customer metadata,ENT-9629
-            # self.load_enterprise_customer()
             self.load_and_process_subscription_licenses()
 
             # Retrieve default enterprise courses and enroll in the redeemable ones
-            self.load_default_enterprise_courses()
-            self.enroll_in_redeemable_default_courses()
+            self.load_default_enterprise_enrollment_intentions()
+            self.enroll_in_redeemable_default_enterprise_enrollment_intentions()
         except Exception as e:  # pylint: disable=broad-exception-caught
             logger.exception("Error loading learner portal handler")
             self.add_error(
                 user_message="An error occurred while loading and processing common learner logic.",
                 developer_message=f"Error: {e}",
             )
 
+    def transform_enterprise_customers(self):
+        """
+        Transform enterprise customer metadata retrieved by self.context.
+        """
+        if enterprise_customer := self.context.data.get('enterprise_customer'):
```

Comment:
nit: `if self.context.enterprise_customer`

Reply:
actually, how about a loop like this?
```python
for customer_record_key in ('enterprise_customer', 'active_enterprise_customer', 'staff_enterprise_customer'):
    if not customer_record := self.context.data.get(customer_record_key):
        continue
    self.context.data[customer_record_key] = self.transform_enterprise_customer(customer_record)
```

Reply:
I like the loop suggestion to make the implementation more DRY. I believe the conditional logic around `if enterprise_customer_users := self.context.data.get('all_linked_enterprise_customer_users'):` will still need to remain, though.

```python
def transform_enterprise_customers(self):
    """
    Transform enterprise customer metadata retrieved by self.context.
    """
    for customer_record_key in ('enterprise_customer', 'active_enterprise_customer', 'staff_enterprise_customer'):
        if not (customer_record := self.context.data.get(customer_record_key)):
            continue
        self.context.data[customer_record_key] = self.transform_enterprise_customer(customer_record)

    if enterprise_customer_users := self.context.data.get('all_linked_enterprise_customer_users'):
        self.context.data['all_linked_enterprise_customer_users'] = [
            self.transform_enterprise_customer_user(enterprise_customer_user)
            for enterprise_customer_user in enterprise_customer_users
        ]
```